portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-download.controller.js

```diff
@@ -5,30 +5,36 @@ const { FILESHARES } = require('../../../constants/file-upload');
 
 /**
  * Fetches metadata about the utilisation report file associated with the
- * specified MongoDB _id
- * @param _id {string} - the MongoDB _id of the report
+ * specified id and bankId
+ * @param bankId {string} - the id of the bank
+ * @param id {number} - the SQL id of the report
  * @returns {Promise<{ filename: string; mimetype: string }>}
  */
-const getUtilisationReportFileMetadata = async (_id) => {
-  const { azureFileInfo } = await api.getUtilisationReportById(_id);
+const getUtilisationReportFileMetadata = async (bankId, id) => {
+  const response = await api.getUtilisationReportById(id);
+  const { azureFileInfo, bankId: reportBankId } = response;
+
+  if (reportBankId !== bankId) {
+    throw new Error(`Failed to find utilisation report with bankId ${bankId} and report id ${id}`);
+  }
 
   if (!azureFileInfo?.filename) {
-    throw new Error(`Failed to get filename for utilisation report with _id '${_id}'`);
+    throw new Error(`Failed to get filename for utilisation report with id '${id}'`);
```

Comment:
Out of curiosity, is there a situation where we would have a 200 response from `getUtilisationReportById` where `azureFileInfo` would be missing either `filename` or `mimetype`? I feel if not, we can just check the response from `api.getUtilisationReportById(id);` and throw a `UtilisationReportNotFound` error at the `api` level, and can bin off a lot of extra code!